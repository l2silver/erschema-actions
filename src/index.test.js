// @flow
jest.mock('redux-batched-actions', ()=>({batchActions: (action)=>action}))
jest.mock('redux-retype-actions', ()=>({retypeAction: (name, action)=>action}))
jest.mock('erschema-redux-immutable/actions/entities', ()=>({
  index: (entityName, entities)=>({entityName, entities})
}))
jest.mock('erschema-redux-immutable/actions/relationships', ()=>({
  index: (entityName, relationships)=>({entityName, relationships})
}))
import Actions from './'
import standardizeEntity from 'erschema-redux-immutable/schemas'

const getId = ()=>Math.floor((Math.random()*100000))
describe('Action', function() {
  const schema = {
    users: standardizeEntity({
      Model: class {},
      properties: ['name', 'id'],
      relationships: [{
        entityName: 'users',
        name: 'friends',
      }]
    }),
    friends: standardizeEntity({
      Model: class {},
      properties: ['name', 'id']
    })
  };
  class ActionWithSchema extends Actions {
    constructor(name){
      super(schema, name)
    }
  }
  const action = new ActionWithSchema('users')
  describe('creates an action instance with access to', function(){
    it('get', function() {
      const entity = {id: 1, name: 'John'}
      const results = action.entities.get({...entity, friends: []})
      expect(results).toMatchSnapshot()
    })
  })
})