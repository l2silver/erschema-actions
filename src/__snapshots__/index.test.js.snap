// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action creates an action instance with access to get 1`] = `
Object {
  "indexEntities": Array [
    Object {
      "entities": Array [
        Object {
          "id": 1,
          "name": "John",
        },
        Object {
          "id": undefined,
        },
      ],
      "entityName": "users",
    },
  ],
  "indexRelationships": Array [
    Object {
      "entityName": "users",
      "relationships": Object {
        "idValuePairs": Array [
          Object {
            "id": "1",
            "value": undefined,
          },
        ],
        "name": "friends",
      },
    },
  ],
}
`;
